react/features/display-name/components/web/DominantSpeakerName.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import { makeStyles } from '@material-ui/core/styles';
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import { getDominantSpeakerParticipant, getLocalParticipant, getPinnedParticipant } from '../../../base/participants';
+import { withPixelLineHeight } from '../../../base/styles/functions.web';
+import { isToolboxVisible } from '../../../toolbox/functions.web';
+import { isLayoutTileView } from '../../../video-layout';
+
+import DisplayNameBadge from './DisplayNameBadge';
+
+const useStyles = makeStyles(theme => {
+    return {
+        badgeContainer: {
+            ...withPixelLineHeight(theme.typography.bodyShortRegularLarge),
+            alignItems: 'center',
+            display: 'flex',
+            justifyContent: 'center',
+            marginBottom: theme.spacing(2),
+            transition: 'margin-bottom 0.3s'
+        },
+        containerElevated: {
+            marginBottom: theme.spacing(7)
+        }
+    };
+});
+
+/**
+ * Component that renders the dominant speaker's name as a badge above the toolbar in stage view.
+ *
+ * @returns {ReactElement|null}
+ */
+const DominantSpeakerName = () => {
+    const classes = useStyles();
+
+    const pinnedParticipant = useSelector(getPinnedParticipant);
+    const dominantSpeaker = useSelector(getDominantSpeakerParticipant);
+    const selectedParticipant = pinnedParticipant ?? dominantSpeaker;
```

Comment:
I'm a little bit confused. Do we try to show the dominant speaker or we actually want to show the participant that is currently on stage?

If we want to show the name of the participant that is currently on stage, the logic should be a little bit more complicated. See the `_electParticipantInLargeVideo` function in `react/features/large-video/actions.any.js`. I believe we store our current selection in redux in `features/large-video`.

Reply:
Awesome tip, thanks!